src/models/connectionProfile.ts

```diff
@@ -152,4 +153,14 @@ export class ConnectionProfile extends ConnectionCredentials implements IConnect
         }
         return choices;
     }
+
+    public setProfile(connectionCredentials: IConnectionInfo): void {
```

Comment:
This is fine, but you could perhaps create a connection profile within this helper and then return that instead of just setting fields, unless you plan to use this helper elsewhere